invideoquiz/public/js/src/config.js

```diff
@@ -6,7 +6,12 @@ var InVideoQuizXBlock = InVideoQuizXBlock || {{}};
     InVideoQuizXBlock.config = InVideoQuizXBlock.config || {{}};
 
     var videoId = '{video_id}';
-    if (videoId) {{
-        InVideoQuizXBlock.config[videoId] = {timemap};
+    try {{        
+        if (videoId) {{
+            InVideoQuizXBlock.config[videoId] = JSON.parse(`{timemap}`.replace(/\'/g,`"`));
+        }}
+    }}
+    catch {{
+        window.alert("There is something wrong with the timemap of the in video quiz XBlock.");
```

Comment:
not sure how can I add `gettext` here, any example? thx

Reply:
@sarina @iamCristYe Honestly, I don't think we tended to do much, if any, frontend i18n (on the Stanford-family of XBlocks), instead opting for translations in python, either via the template or via `context` variables.

Since we're definitely already doing Python-level i18n in this block, I'd probably first suggest translating it on the backend, since we know that part is already hooked up; one less new thing to worry about wiring up :shrug: 

_That_ _said_, I don't think we want to catch this error here, at display/runtime. Instead, we should be catching this error during validation/saving.

This XBlock uses the `StudioEditableXBlockMixin` [1], so I _think_, based on the docs [1],  it should be as easy as adding a `validate_timemap` method to `InVideoQuizXBlock`.

That gives us the best of both worlds; a tighter feedback loop (display the error to the instructor when they misconfigure the problem, instead of to the user when they load the problem), as well as easier, backend validation/error messages (since we're validating it on the backend anyway).


- [1] https://github.com/edx/xblock-utils#studioeditablexblockmixin